pp/go/cppbridge/metrics.go

```diff
@@ -1,27 +1,57 @@
 package cppbridge
 
 import (
+	"github.com/prometheus/client_golang/prometheus"
 	dto "github.com/prometheus/client_model/go"
-	"google.golang.org/protobuf/proto"
 )
 
-func CppMetrics(f func(metric *dto.Metric) bool) {
+type CppMetric struct {
+	descriptor *prometheus.Desc
+	metric     *dto.Metric
+}
+
+func (m *CppMetric) Desc() *prometheus.Desc {
+	return m.descriptor
+}
+
+func (m *CppMetric) Write(out *dto.Metric) error {
+	out.Label = m.metric.Label
```

Comment:
`out` may be reused with go code. So it may fail on realloc labels slice. I think it is better make copy here

Reply:
Looks like it's OK. Other collectors replace labels slice whole